### ISSUE TITLE
Not possible to set response status on Undertow

### ISSUE THREAD

**Issue by rstoyanchev** [spring-projects/spring-reactive#119]
It seems that the [call to getResponseChannel()](https://github.com/spring-projects/spring-reactive/blob/0a96b7a7f820e9ab6d5e4abd574c68272a6b77f9/src/main/java/org/springframework/http/server/reactive/UndertowHttpHandlerAdapter.java#L70) in turn calls `startResponse()` internally which commits the response. The `HttpServerExchange` has various setters so perhaps the idea is to call those.

**Comment by rstoyanchev** [comment]
````
2016-06-22 09:13:48,884 DEBUG [org.springframework.web.server.adapter.HttpWebHandlerAdapter] <XNIO-1 I/O-2> - Could not complete request
java.lang.IllegalStateException: UT000002: The response has already been started
	at io.undertow.server.HttpServerExchange.setStatusCode(HttpServerExchange.java:1352)
	at org.springframework.http.server.reactive.UndertowServerHttpResponse.setStatusCode(UndertowServerHttpResponse.java:78)
	at org.springframework.web.reactive.result.method.annotation.ResponseEntityResultHandler.handleResult(ResponseEntityResultHandler.java:99)
	at org.springframework.web.reactive.DispatcherHandler.handleResult(DispatcherHandler.java:125)
	at org.springframework.web.reactive.DispatcherHandler.lambda$handle$2(DispatcherHandler.java:112)
````

**Comment by poutsma** [comment]
Fixed in ca475e1
